aten/src/ATen/native/native_functions.yaml

```diff
@@ -2415,6 +2416,7 @@
     SparseCsrCPU, SparseCsrCUDA: empty_like_sparse_csr
     NestedTensorCPU, NestedTensorCUDA: empty_like_nested
   autogen: empty_like.out
+  tags: core
```

Comment:
Why would `empty_like` require `as_strided`? Cann't we use the reference which decomposes to `aten.empty_permuted` ?

The same applies to `full_like` if #110234 gets merged.

Reply:
@peterbell10 that is definitely an oversight on my part, my apologies. In fact I see that `empty_like` is already being decomposed as part of the core ATen decomposition table, so for `empty_like` this is definitely an erroneous classification.

I've created https://github.com/pytorch/pytorch/pull/110924 to fix this, please take a look. As part of that diff, I also registered the reference implementation of `full_like` as a core decomposition, with some slight changes to the implementation to use `torch` ops instead of `prims`.

For my own understanding, is it possible that the input tensor has weird strides that can't translate cleanly to a permutation?

Reply:
Copy pasting from my comment on the PR:

> Okay, nevermind. I see why full_like cannot be registered as a decomposition this way (it is due to recursion with fill.Scalar decomposition. So this diff will just remove empty_like as a core decomposition, and when https://github.com/pytorch/pytorch/pull/110234 lands we can decompose full_like as well.

Reply:
> For my own understanding, is it possible that the input tensor has weird strides that can't translate cleanly to a permutation?

`torch.empty_like` in eager always returns a dense tensor and only the ordering of dimensions is influenced by the "like" tensor argument.